### ISSUE TITLE
Behavior of quotes in markdown type of links

### ISSUE THREAD

**Issue by couet** [doxygen/doxygen#10573]
**Describe the bug**
In my code, I have something like this:

```
- [The "BAR" and "HBAR" options](\ref HP10)
```
In the past (until version 1.9.8 at least)  the text between the squared bracket was made a link to the reference `HP10`

Now, with doxygen 1.11, the quotes in my text make the result wrong. I get something like:

<img width="262" alt="Screenshot 2024-01-19 at 11 36 50" src="https://github.com/doxygen/doxygen/assets/4697738/2c3c05f1-c664-46b0-9c9e-240d4a8648b8">

The first double quote disappeared a new one appeared after **options** and only **The** is made a web link.

I noticed that if I escaped the double quotes with `\` then it worked. 

Is it a new behavior or a bug? Should we escape all the double quotes when they appeared in squared brackets?

**Version**
1.11.0

**Comment by albert-github** [comment]
Doxygen 1.11.0 is the current master version and as this is a moving target please specify the full version (`doxygen -v Doxyfile`).

With the version: 1.11.0 (db523de7141a3f138c4f0a723249fd1f8816a54a) I could reproduce the issue.
Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/13988172/example.tar.gz)

**Comment by couet** [comment]
```
% doxygen -v Doxyfile
1.11.0 (db523de7141a3f138c4f0a723249fd1f8816a54a)
```
So it is a bug ?

**Comment by albert-github** [comment]
> So it is a bug ?

Most likely, see the label I added. It has to be checked what the reason of the change in behavior is.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10575

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
